int32[] pos_x
int32[] pos_y
int32[] target_id
int32 id